packages/core/parcel-bundler/src/transforms/babel/jsx.js

```diff
@@ -24,9 +24,17 @@ function createJSXRegexFor(dependency) {
  * package.json is empty or missing yet and therefore pragma cannot
  * be determined based on pkg.dependencies / pkg.devDependencies
  */
+const cacheJsxRegexFor = {};
```

Comment:
@devongovett, hope that goes in direction of your proposal. Regexes are lazily created in case needed and then cached (similarly to ahead of time creation).